certbot-dns-dnsimple/snap/snapcraft.yaml

```diff
@@ -16,10 +16,23 @@ parts:
         snapcraftctl set-version `grep ^version $SNAPCRAFT_PART_SRC/setup.py | cut -f2 -d= | tr -d "'[:space:]"`
     build-environment:
       - EXCLUDE_CERTBOT_DEPS: "True"
+  certbot-version:
```

Comment:
Did you run `snapcraft clean` before doing that?

Reply:
Ah my problem here was I didn't know how to actually trigger a refresh. Once I did that, it fails without this so you can ignore this thread.

(I was also confused because the Certbot snap has nothing like this for the interface it reads from so this which made some of this seem superfluous to me. The reason the Certbot snap works though is because it makes use of its lack of confinement rather than reading from the interface like normal. See [this](https://github.com/certbot/certbot/blob/05dbda4b519daa1db08a0cf0a730b9240e37c65b/certbot.wrapper#L14)).

Reply:
> The reason the Certbot snap works though is because it makes use of its lack of confinement rather than reading from the interface like normal.

Oh I am unfortunately well aware of this. When I went looking for an example of how to do this successfully, I thought at first that we would have one internally, then was disappointed to find that we didn't.